### PR TITLE
Merge 6.0.0 Dev Release to Stable

### PR DESCRIPTION
**Pull request by robrbecker**
The dev branch when merged will be the 6.0 release.

## Overview of changes:
- There's a single entrypoint now `package:github/github.dart`
- `createGithubClient(...)` has been removed. Just create a GitHub object directly now.
- `findAuthenticationFromEnvironment` now works in both server/flutter and web environments
  - On the web, it will check the query string first, then session storage
- all static methods are now factory constructors
- Use JsonSerializable everywhere
  - fromJSON is now fromJson everywhere
  - toJSON is now toJson everywhere
- removed deprecated items
- renamed some fields with ID at the end to be Id
- most model constructors now have named parameters for all properties

### CODE REVIEW

**Comment by robrbecker** [comment]
@johnpryan @kevmoo @pq @axel-op I released a dev release of this branch. If you have a moment, would you give it a try in your projects? https://pub.dev/packages/github/versions/6.0.0-dev.1.0
There are known breaking changes, but hopefully it shouldn't be too much. File issues or comment on this PR.

**Comment by pq** [comment]
Looks good here after a quick test.  Thanks for the heads-up!

If this is the API moving forward, I'm inclined to update my deps to use `6.0.0-dev.1.0`.  Does that sound like a good idea or premature?

**Comment by robrbecker** [comment]
Probably a bit premature still. I'd like to spend a week polishing things
up a bit more and testing.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,18 @@
...
+- fromJSON is now fromJson everywhere
+- toJSON is now toJson everywhere
+- Use JsonSerializable everywhere
+- removed deprecated items
+- renamed some fields with ID at the end to be Id
+- most model constructors now have named parameters for all properties
+
 ## v5.5.0
 
 - Provide a single platform independent import `import 'package:github/githb.dart';`
```
Comment (axel-op) [inline CHANGELOG.md]:
```suggestion
- Provide a single platform independent import `import 'package:github/github.dart';`
```

**Pull request merged**